NewKernel_d/include/CGAL/NewKernel_d/Kernel_d_interface.h

```diff
@@ -175,6 +176,16 @@ template <class Base_> struct Kernel_d_interface : public Base_ {
 	    return typename Get_functor<Base, Squared_circumradius_tag>::type(this->kernel())(b,e);
 	  }
 	};
+	struct Compute_squared_radius_smallest_orthogonal_sphere_d : private Store_kernel<Kernel> {
+	  typedef Kernel R_; // for the macro
+	  CGAL_FUNCTOR_INIT_STORE(Compute_squared_radius_smallest_orthogonal_sphere_d)
+	  typedef FT result_type;
+	  template<class I> FT operator()(I b, I e)const{
+	    typename Get_functor<Base, Point_weight_tag>::type pw(this->kernel());
+	    typename Get_functor<Base, Power_center_tag>::type pc(this->kernel());
+	    return pw(pc(b,e));
```

Comment:
Is not there a better implementation that does not compute the center, like `squared_radius_smallest_orthogonal_sphereC3`?

Reply:
When the number of spheres is dim+1, yes. When it is less, I don't know. `squared_radius_smallest_orthogonal_sphereC3` only works with 4 points. My priority is to have the functionality, but indeed I don't consider my implementation as a high quality one, and I'd be happy if someone wants to contribute a better one later.

Reply:
That is right.

By the way, are you sure you want to hard-code that `Compute_squared_radius_smallest_orthogonal_sphere_d` is so general that it can be called with any number of independent points? Should not it be restricted to dim+1, and then create another one for the general case?

Reply:
"smallest" in the name is there because we may have a small number of points. A version only for the maximal case would not need "smallest" in its name: there is only 1 sphere in that case. `ComputeSquaredRadiusSmallestOrthogonalSphere_3` you mentioned also handles 1-4 points, that's what is needed for an alpha-complex.

Reply:
You are right... Good. Maybe later we can imagine add overloads and have one restricted to dim+1. I wonder if contracts or concepts can add sfinae-like restriction on overloads, based on `std::distance(first, last)`. Or maybe just a `if constexpr`.

Reply:
Or a plain `if`, like I already have inside Construct_circumcenter...